### ISSUE TITLE
Adding a large number of rows breaks the table with a maximum stack exceeded error.

### ISSUE THREAD

**Issue by jansiegel** [handsontable/handsontable#7840]
### Description
When adding a large number of rows (exactly 62522+ rows) to an instance, it crashes with the `Maximum call stack size exceeded` error.

Seems to happen on `8.0.0`+.

### Steps to reproduce

1. Go to https://jsfiddle.net/ef7chuk9/.
2. Check the console.

### Demo
https://jsfiddle.net/ef7chuk9/

### Your environment
* Handsontable version: 8.3.2
* Browser Name and version: Latest chrome
* Operating System: MacOS Catalina

**Comment by wojciechczerniak** [comment]
This method https://github.com/handsontable/handsontable/blob/fa685dc8c194c0269b893d202580ce7091ed8322/src/dataMap.js#L582-L588
Is transpiled to `splice.apply`
```ts
    key: "spliceData",
    value: function spliceData(index, amount) {
      for (var _len3 = arguments.length, elements = new Array(_len3 > 2 ? _len3 - 2 : 0), _key3 = 2; _key3 < _len3; _key3++) {
        elements[_key3 - 2] = arguments[_key3];
      }

      var continueSplicing = this.instance.runHooks('beforeDataSplice', index, amount, elements);

      if (continueSplicing !== false) {
        var _this$dataSource;

        (_this$dataSource = this.dataSource).splice.apply(_this$dataSource, [index, amount].concat(elements));
      }
    }
```

> But beware: by using apply this way, you run the risk of exceeding the JavaScript engine's argument length limit. The consequences of applying a function with too many arguments (that is, more than tens of thousands of arguments) varies across engines. (The JavaScriptCore engine has hard-coded argument limit of 65536.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply#using_apply_and_built-in_functions

**Comment by adrianszymanski89** [comment]
This issue is still replicable in v. 12.0.1:

https://jsfiddle.net/aszymanski/su87k1v3/

**Comment by semlak** [comment]
I'm having similar problem that I believe is same root issue. With a table with a large number of rows that is otherwise displaying okay, with column filters enabled, you hit the drop down for the column to select a filter, and the menu can't open because it hits a maximum callstack size. I think it is from when computing the values that it can filter on.
The method `getDataAtCol`. 
```
 this.getDataAtCol = function (column) {
    var _ref13;

    return (_ref13 = []).concat.apply(_ref13, _toConsumableArray(datamap.getRange(new CellCoords(0, column), new CellCoords(tableMeta.data.length - 1, column), datamap.DESTINATION_RENDERER)));
  };
```
Here's a jsfiddle set up to generate error. https://jsfiddle.net/semlak/cv0geoLh/11/

**Comment by AMBudnik** [comment]
Thank you for sharing the report, @semlak 

In the Author's case we get the following error

```js
handsontable.full.js:95279 Uncaught RangeError: Maximum call stack size exceeded
    at DataMap.spliceData (handsontable.full.js:95279:53)
    at DataMap.createRow (handsontable.full.js:94996:23)
    at Object.alter (handsontable.full.js:83562:27)
    at Core.alter (handsontable.full.js:85737:10)
    at ?editor_console=true:122:8
```
while with the dropdown menu (your case) we get

```js
handsontable.full.js:85958 Uncaught RangeError: Maximum call stack size exceeded
    at Core.getDataAtCol (handsontable.full.js:85958:33)
    at ValueComponent._getColumnVisibleValues (handsontable.full.js:158666:44)
    at ValueComponent.reset (handsontable.full.js:158631:55)
    at ValueComponent.setState (handsontable.full.js:158484:12)
    at ValueComponent.restoreState (handsontable.full.js:43937:14)
    at handsontable.full.js:155864:19
    at Array.forEach (<anonymous>)
    at Filters.restoreComponents (handsontable.full.js:155859:18)
    at Filters.onAfterDropdownMenuShow (handsontable.full.js:155877:12)
    at Core.<anonymous> (handsontable.full.js:155509:23)
```

The root might be the same. But still, those are two different errors (dropped by different places in the code). 

At the moment we can keep this case as a 2nd scenario. But during the development, we may need to split those two issues.

**Comment by AMBudnik** [comment]
@semlak 's case is definitely related to `filters` issue and the large dataset. When we have `125k` rows and open a dropdownMenu with filters we get the `Maximum call stack size exceeded` error at `getDataAtCol()`. Demo http://jsfiddle.net/muL3bqke/3/

**Comment by AMBudnik** [comment]
Hey @semlak I've decided to create a new issue for our two cases at https://github.com/handsontable/handsontable/issues/9772
I will notify you upon fix.

**Comment by semlak** [comment]
Thanks!

**Comment by warpech** [comment]
Moving back to the "Review" pipeline, because QA discovered a problem in Safari (described in https://github.com/handsontable/handsontable/pull/9855#issuecomment-1246837359)

**Comment by aninde** [comment]
Fixed by v12.1.3 https://jsfiddle.net/tzmgky17/
 It works on the LTS Chrome 105.0.5195.125, Safari 15.6 (17613.3.9.1.5), Firefox 104, and Edge 105.0.1343 on Windows 10 and macOS 12.5. RTL mode also works.

**Comment by AMBudnik** [comment]
I'm closing this issue as solved in Handsontable v12.1.3.
